nats-experimental/src/lib.rs

```diff
@@ -121,8 +124,18 @@ pub struct Connection {
 }
 
 impl Connection {
-    pub async fn connect(addr: impl ToSocketAddrs) -> Result<Connection, io::Error> {
-        let tcp_stream = TcpStream::connect(addr).await?;
+    pub async fn connect(addrs: impl IntoServerList) -> Result<Connection, io::Error> {
```

Comment:
Lets prefer explicit over implicit, unless we've made an explicit deliberate choice to do implicit conversion.
 ```suggestion
    pub async fn connect<A: ToServerAddrs>(addrs: A) -> Result<Connection, io::Error> {
```

Reply:
agree. Despite not liking abbreviations in Rust :). Fixef.